Shared/Samples/Filter building scene layer/FilterBuildingSceneLayerView.swift

```diff
@@ -0,0 +1,208 @@
+    @State private var settingsAreVisible = false
+    
+    /// The available floors to show in the settings.
+    @State private var availableFloors: [String] = []
+    
+    /// The selected floor from the floor picker in the settings.
+    @State private var selectedFloor: String = .allFloorsLabel
+    
+    /// The group sublayers used to build the visibility toggles in the settings.
+    @State private var groupSublayers: [BuildingGroupSublayer] = []
+    
+    /// The error shown in the error alert.
+    @State private var error: (any Error)?
+    
+    /// The point where the user has tapped on the screen.
+    @State private var tapPoint: CGPoint?
+    
+    /// The building sublayer that has a feature selected.
+    ///
+    /// This is used to clear the feature selection.
+    @State private var selectedSublayer: BuildingComponentSublayer?
+    
+    /// The popup to be shown as the result of the layer identify operation.
+    @State private var popup: Popup? {
+        didSet { showPopup = popup != nil }
+    }
+    
+    /// A Boolean value specifying whether the popup view should be shown or not.
+    @State private var showPopup = false
+    
+    var body: some View {
+        LocalSceneViewReader { proxy in
+            LocalSceneView(scene: scene)
+                .onSingleTapGesture { screenPoint, _ in
+                    tapPoint = screenPoint
+                }
+                .task {
+                    do {
+                        try await scene.load()
+                        
+                        guard let buildingSceneLayer = scene.operationalLayers.first as? BuildingSceneLayer else { return }
+                        
+                        try await buildingSceneLayer.load()
+                        self.buildingSceneLayer = buildingSceneLayer
+                        
+                        // Gets the full model which contains all the sublayers.
+                        guard let fullModelSublayer = buildingSceneLayer.sublayers
+                            .first(where: { $0.name == "Full Model" }) as? BuildingGroupSublayer else { return }
+                        
+                        groupSublayers = fullModelSublayer.sublayers
+                            .compactMap { $0 as? BuildingGroupSublayer }
+                        
+                        // Gets the attribute statistics to get the floor
+                        // information.
+                        let statistics = try await buildingSceneLayer.statistics
+                        
+                        // Gets the floor statistics.
+                        guard let floorStatistics = statistics[.floorFieldKey] else { return }
+                        
+                        // Gets all the available floors and sorts the floors
+                        // from top to bottom floor.
+                        availableFloors = floorStatistics.mostFrequentValues.sorted { $0 > $1 } + [.allFloorsLabel]
```

Comment:
This is a really long closure. I recommend extracting these lines into a method and then calling that method from this do block.